pkg/wallet/account.go

```diff
@@ -128,6 +129,9 @@ func NewAccount() (*Account, error) {
 
 // SignTx signs transaction t and updates it's Witnesses.
 func (a *Account) SignTx(t *transaction.Transaction) error {
+	if t.Nonce == 0 {
+		t.Nonce = rand.Uint32()
```

Comment:
I don't think `SignTx` is a right place for generating nonce. Maybe we could do this while creating transaction?

Reply:
We don't have separate NewTransaction method, so I added nonce generation to `NewContractTx()`, `NewInvocationTx()`, ... and `claimGas`, `transferAsset` methods of `cli/wallet/wallet.go`. However, we have a lot of places, where we use `transaction.Transaction{...}`, should it be added to all of these places?